llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -405,6 +412,17 @@ static bool isEqualImpl(SimpleValue LHS, SimpleValue RHS) {
            LII->getArgOperand(1) == RII->getArgOperand(0);
   }
 
+  if (auto *LGEP = dyn_cast<GetElementPtrInst>(LHSI)) {
+    auto *RGEP = cast<GetElementPtrInst>(RHSI);
+    if (LGEP->getOperand(0) != RGEP->getOperand(0))
+      return false;
+    auto &DL = LGEP->getModule()->getDataLayout();
+    APInt LOffset(DL.getIndexTypeSizeInBits(LGEP->getType()), 0);
+    APInt ROffset(DL.getIndexTypeSizeInBits(LGEP->getType()), 0);
```

Comment:
nit: Avoid duplicate getIndexTypeSizeInBits() call (can't have different bitwidths if the pointer operand is the same)